src/internal/oscal/generate.go

```diff
@@ -292,9 +299,49 @@ func generateModelTypes(obj map[string]interface{}, structId string, structName
 	return formattedStructName
 }
 
+// generateOscalComponentDocumentStruct generates the top-level struct for OSCAL data models.
+func generateOscalComponentDocumentStruct(oscalSchema map[string]interface{}, pkgName string, tags []string) string {
+	// Check if there is a top-level 'required' field.
+	if componentDefinitionField := oscalSchema["required"]; componentDefinitionField != nil {
```

Comment:
What are your current thoughts for basing the creation of the struct on the required field? Given that the schema is likely well known - this won't be an issue - but it does change the precedent of a "required" field from something that can be empty to something that would stop creation of the struct field. 

No issue here currently - but I am still thinking through extensible workflows. If we think about a possible future scenario where we provide more than one schema (ie component-definition and system-security-plan) - how would we accomplish the creation of this struct? Would it be a function called multiple times or a single function that can iterate over some object?

A struct in this code that houses this information might also provide some helpers later.

Reply:
> What are your current thoughts for basing the creation of the struct on the required field? Given that the schema is likely well known - this won't be an issue - but it does change the precedent of a "required" field from something that can be empty to something that would stop creation of the struct field.

My thinking was that the top-level `required` field is consistent across all of the OSCAL schemas, so it should give us a predictable place to get any of the top-level fields from.

In this case, the struct wouldn't be created only if the top-level `required` field is empty, which would indicate an invalid schema, and we shouldn't generate a struct from an invalid schema.

> No issue here currently - but I am still thinking through extensible workflows. If we think about a possible future scenario where we provide more than one schema (ie component-definition and system-security-plan) - how would we accomplish the creation of this struct? Would it be a function called multiple times or a single function that can iterate over some object?

I imagine there are many ways we could accomplish this. I also think once we add other schemas, there will probably be a pretty significant refactor/rewrite of our current logic, and my assumption is that generating this top-level struct, even with multiple schemas, will likely be one of the more simple challenges we face compared to the rest of the logic we will have to put together.

> A struct in this code that houses this information might also provide some helpers later.

Are you saying you would like to have the generated struct look like the one you [mentioned](https://github.com/defenseunicorns/go-oscal/issues/10#issuecomment-1421362765) in the issue?

Reply:
> Are you saying you would like to have the generated struct look like the one you https://github.com/defenseunicorns/go-oscal/issues/10#issuecomment-1421362765 in the issue?

Yes, that would be an example of what the generated struct would look like. 

This comment:

> A struct in this code that houses this information might also provide some helpers later.

Was referencing a struct that would be within `generate.go` and allow for the data to be co-located across models. That way, if we processed multipel models, we could aggregate the data and then have a single place to begin looking at de-duplication - rather than having to track multiple transient objects for each model.